### ISSUE TITLE
Trying to download large email (>5MB) throws  when options http2: true

### ISSUE THREAD

**Issue by lopezvit** [googleapis/google-api-nodejs-client#3109]
#### Environment details

  - OS: Linux / Macos
  - Node.js version: v18.10.0
  - npm version:  8.19.2
  - `googleapis` version: 105/109.0.1

#### Steps to reproduce

  1. Get a large email from Google API with the options {http2: false}
  2. The function returns the following error:
  
```
Could not download email 17d03869ffcd2de1 Error: unexpected end of file
    at Zlib.zlibOnError [as onerror] (node:zlib:189:17)
    at Zlib.callbackTrampoline (node:internal/async_hooks:130:17) {
  errno: -5,
  code: 'Z_BUF_ERROR'
}

```

Making the same request using the [Try this API](https://developers.google.com/gmail/api/reference/rest/v1/users.messages/get?apix_params=%7B%22userId%22%3A%22me%22%2C%22id%22%3A%2217d03869ffcd2de1%22%2C%22format%22%3A%22raw%22%7D)

Making the same request with the {http2: false} option works as well.

Possibly related with:
#3108

**Comment by wdanilo** [comment]
I have also observed that. As http2 should be used now instead of batch calls, this is pretty critical thing.

**Comment by transitive-bullshit** [comment]
In your steps to reproduce, I think you mean: `{http2: true}`

**Comment by lopezvit** [comment]
yeah @transitive-bullshit , of course, in the steps to reproduce it should be the way you are describing it. I just meant, that I tried with `{http2: false}` and then it worked.

Hard to believe that is about the be the 2 year anniversary of this issue, and there is not answer.

Please, if any of you find that you are having the same issue, give a thumb's up 👍  as a reaction to the original post, so maybe it would be taken more into account by the developer team.
